src/main/kotlin/no/nav/syfo/service/SendVarselService.kt

```diff
@@ -21,47 +25,58 @@ class SendVarselService(
     val beskjedKafkaProducer: BeskjedKafkaProducer,
     val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
     val narmesteLederService: NarmesteLederService,
-    val accessControl: AccessControl,
+    val accessControlService: AccessControlService,
     val urlEnv: UrlEnv,
     val syfoMotebehovConsumer: SyfoMotebehovConsumer,
     val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
+    val journalpostdistribusjonConsumer: JournalpostdistribusjonConsumer,
+    val dokarkivService: DokarkivService,
 ) {
-    private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.service.SendVarselService")
+    private val LOG: Logger = LoggerFactory.getLogger("no.nav.syfo.service.SendVarselService")
+
     val WEEKS_BEFORE_DELETE_AKTIVITETSKRAV = 2L
 
     suspend fun sendVarsel(pPlanlagtVarsel: PPlanlagtVarsel): String {
```

Comment:
Jeg har tenkt en stund at denne klassen begynner å bli veldig bloated. Et mulig design kunne være å opprette `AktivitetskravVarselService` og `MerVeiledningVarselService` med metoder som `sendVarselTilNarmesteLeder()` og `sendVarselTilArbeidstaker()` etter modellen `MotebehovVarselService`. Det vil ikke fungere så godt for akkurat Svar møtebehov, siden vi her gjør et kall til syfomotebehov som igjen sender en melding tilbake til esyfovarsel, men håndteringen av Svar møtebehov vil uansett fjernes når vi ferdigstiller kandidatliste-funksjonaliteten.

Det er ikke sikkert vi skal refaktorere den i denne pr, men kanskje i en senere PR. Vil gjerne høre hva andre tenker om dette.

Reply:
@kaja-nav Jeg er enig med tankene dine angående refaktorering. Det blir lettere å forstå de forskjellige varselkontekstene 👍 men det blir nok en annen PR ja for å unngå at denne blir for bloated 🙂